### PR TITLE
.Net: [MEVD] Test cleanup pack

### PR DESCRIPTION
**Pull request by roji**
This is an (almost) entirely test-only cleanup (I plan to do a bit more later).

* Remove BasicQueryTests, rename BasicFilterTests to FilterTests and move out to the root
    * BasicFilterTests and BasicQueryTests both run the same filter scenarios, the former via SearchAsync and the latter via filtering GetAsync. Although in theory, some filter query construct could work in one but not in the other, in practice no such example exists in the 12 providers we have (all databases use the same filter language/engine for both APIs. So we had needless duplication for all the various test exceptions/customizations, which had to be overridden in exactly the same way in both test suites.
    * At least for now, decided to remove BasicQueryTests as it adds maintenance burden without any real coverage; coverage (included missing coverage) specifically for filtered GetAsync was added separately (e.g. we weren't testing ordering, skip/take with filtered get).
    * If we do want to add back support for all query constructs (I'd want to have a good reason to do so), I'd do that within FilterTests: just like each scenario already exercises both static and dynamic models, it could run GetAsync as well.
* Renamed CollectionConformanceTests to CollecitonManagementTests and moved it out to the root.
    * In general, this PR removes "Conformance" from the test type names.
* Big cleanup/refactor of all the so-called CRUD tests.
    * We now have a ModelTests folder with a test suite for each different model; BasicModelTests covers Get/Upsert/Delete (and their batching variants) for the basic scenario (strongly-typed model with data properties and a single vector property). Alongside that we have DynamicModelTests, NoDataModelTests, NoVectorModelTests (I'll be added MultipleVectorsModelTests).
    * These tests no longer use random GUIDs for the collection name (difficult to debug, the collection never gets cleaned up if the test crashes/is stopped in the middle), but rather has fixed names per test suite.
    * The collection is created once per test suite, but cleared and reseeded between each test run, providing full isolation (previously we had no isolation, and documented each record in the seed data as belonging to this scenario or that - which is brittle).
    * Many other small improvements and cleanup.

Part of #12507

### CODE REVIEW

**Comment by westey-m** [comment]
> If we do want to add back support for all query constructs (I'd want to have a good reason to do so), I'd do that within FilterTests: just like each scenario already exercises both static and dynamic models, it could run GetAsync as well.

@roji, it might be worth considering covering both when we publish the tests as a nuget for external implementations to verify conformance, although I agree it's an edge for a DB to require different mapping code for Search and Get

File: dotnet/test/VectorData/MongoDB.ConformanceTests/MongoTestSuiteImplementationTests.cs [hunk 1]
```diff
@@ -12,6 +12,6 @@ public class MongoTestSuiteImplementationTests : TestSuiteImplementationTests
     [
         typeof(VectorSearchDistanceFunctionComplianceTests<>),
         typeof(VectorSearchWithFilterConformanceTests<>),
-        typeof(DynamicDataModelConformanceTests<>),
+        // typeof(DynamicModelTests<>),
```
Comment (westey-m) [inline dotnet/test/VectorData/MongoDB.ConformanceTests/MongoTestSuiteImplementationTests.cs]: Can we just remove this?
```suggestion
```

File: dotnet/test/VectorData/MongoDB.ConformanceTests/MongoTestSuiteImplementationTests.cs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 
 using VectorData.ConformanceTests;
-using VectorData.ConformanceTests.CRUD;
+// using VectorData.ConformanceTests.CRUD;
```
Comment (westey-m) [inline dotnet/test/VectorData/MongoDB.ConformanceTests/MongoTestSuiteImplementationTests.cs]:
```suggestion
```

**Comment by roji** [comment]
> it might be worth considering covering both when we publish the tests as a nuget for external implementations to verify conformance, although I agree it's an edge for a DB to require different mapping code for Search and Get

Yeah. We can add this, but I'd just like to first at least see a case or two before we do, is all - in the 12 providers we have, we haven't had such a case (and it's really weird to imagine a case that would).
